packages/backend/src/app/workers/flow-worker/flow-worker.ts

```diff
@@ -22,6 +23,36 @@ import { collectionService } from "../../collections/collection.service";
 import { engineHelper } from "../../helper/engine-helper";
 import { createRedisLock } from "../../database/redis-connection";
 import { captureException, logger } from "../../helper/logger";
+import { packageManager, PackageManagerDependencies } from "../../helper/package-manager";
+
+const extractPieceDependencies = (flowVersion: FlowVersion): PackageManagerDependencies => {
+    const pieceDependencies: PackageManagerDependencies = {};
+
+    const trigger = flowVersion.trigger;
+    if (trigger && trigger.type == TriggerType.PIECE) {
+        const packageName = `@activepieces/piece-${trigger.settings.pieceName}`;
+        const packageVersion = trigger.settings.pieceVersion;
+        pieceDependencies[packageName] = packageVersion;
+    }
+
+    let action: Action = flowVersion.trigger?.nextAction;
+    while (action) {
+        if (action.type === ActionType.PIECE) {
+            const packageName = `@activepieces/piece-${action.settings.pieceName}`;
+            const packageVersion = action.settings.pieceVersion;
+            pieceDependencies[packageName] = packageVersion;
+        }
+
```

Comment:
Leaving comment there, as this will break when we merge the branches.

Reply:
Use `flowHelper.getAllSteps` , this logic no longer works

Reply:
ok